src/widget/container/mod.rs

```diff
@@ -0,0 +1,158 @@
+mod alignment;
+mod border;
+mod sizing;
+
+pub use alignment::Alignment;
+pub use border::Border;
+pub use sizing::Sizing;
+
+use super::{IntrinsicSize, Widget};
+use alloc::{boxed::Box, vec::Vec};
+use embedded_graphics::{
+    prelude::*,
+    primitives::{CornerRadii, PrimitiveStyleBuilder, Rectangle, RoundedRectangle},
+};
+
+pub struct Options<Display>
+where
+    Display: DrawTarget,
+{
+    pub alignment: Alignment,
+    pub background_color: Option<Display::Color>,
+    pub border: Option<Border<Display::Color>>,
+    pub children: Vec<Box<dyn Widget<Display>>>,
+    pub corner_radii: Option<CornerRadii>,
+    pub height: Option<u32>,
+    pub width: Option<u32>,
+}
+
+impl<Display> Default for Options<Display>
+where
+    Display: DrawTarget,
+{
+    fn default() -> Self {
+        Self {
+            alignment: Default::default(),
+            background_color: Default::default(),
+            border: Default::default(),
+            corner_radii: Default::default(),
+            width: Default::default(),
+            height: Default::default(),
+            children: Default::default(),
+        }
+    }
+}
+
+pub struct Container<Display>
+where
+    Display: DrawTarget,
+{
+    options: Options<Display>,
+}
+
+impl<Display> Container<Display>
+where
+    Display: DrawTarget,
+{
+    pub fn new(options: Options<Display>) -> Self {
+        Self { options }
+    }
+
+    fn content_size(&self) -> IntrinsicSize {
+        self.options
+            .children
+            .iter()
+            .fold(IntrinsicSize::none(), |size, widget| {
+                let widget_size = widget.intrinsic_size();
+
+                let width = match (size.width, widget_size.width) {
+                    (Some(width), Some(widget_width)) => Some(width.max(widget_width)),
+                    (Some(width), None) => Some(width),
+                    (None, Some(widget_width)) => Some(widget_width),
+                    (None, None) => None,
+                };
+
+                let height = match (size.height, widget_size.height) {
+                    (Some(height), Some(widget_height)) => Some(height + widget_height),
+                    (Some(height), None) => Some(height),
+                    (None, Some(widget_height)) => Some(widget_height),
+                    (None, None) => None,
+                };
```

Comment:
There is probably a less annoying way of doing this using the `Option` convenience functions, but it wasn't obvious to me what it is.